static/usr/lib/core18/run-snapd-from-snap

```diff
@@ -21,14 +21,17 @@ run_on_unseeded() {
     trap "umount $TMPD; rmdir $TMPD" EXIT
     mount "$SEED_SNAPD" "$TMPD"
 
-    # snapd will write all the needed snapd.{service,socket}
-    # units and restart once it seeded the snapd snap
-    "$TMPD"/usr/lib/snapd/snapd
-
-    # ensure the snapd.socket gets restarted after seeding, the
-    # snapd binary in seeding runs without systemd sockets and
-    # will delete its the socket files it created on exit.
-    systemctl restart snapd.socket || true
+    # snapd will write all its needed snapd.{service,socket}
+    # units and restart once it seeded the snapd snap. We create
+    # a systemd socket unit so that systemd own the socket, otherwise
+    # the socket file would be removed by snapd on exit and the snapd.seeded
+    # service will fail because it has nothing to talk to anymore.
+    systemd-run --unit=snapd-seeding --service-type=notify --socket-property ListenStream=/run/snapd.socket --socket-property ListenStream=/run/snapd-snap.socket "$TMPD"/usr/lib/snapd/snapd
```

Comment:
I think we want `--collect` as well, otherwise the temporary socket and service units will stay behind.

Reply:
This may not be a bad thing for inspecting things in case we need to debug something?